probe/kubernetes/client.go

```diff
@@ -62,35 +64,78 @@ func runReflectorUntil(r *cache.Reflector, resyncPeriod time.Duration, stopCh <-
 	go wait.Until(loggingListAndWatch, resyncPeriod, stopCh)
 }
 
+// ClientConfig establishes the configuration for the kubernetes client
+type ClientConfig struct {
+	Interval             time.Duration
+	CertificateAuthority string
+	ClientCertificate    string
+	ClientKey            string
+	Cluster              string
+	Context              string
+	Insecure             bool
+	Kubeconfig           string
+	Password             string
+	Server               string
+	Token                string
+	User                 string
+	Username             string
+}
+
 // NewClient returns a usable Client. Don't forget to Stop it.
-func NewClient(addr string, resyncPeriod time.Duration) (Client, error) {
-	var config *restclient.Config
-	if addr != "" {
-		config = &restclient.Config{Host: addr}
-	} else {
-		// If no API server address was provided, assume we are running
+func NewClient(config ClientConfig) (Client, error) {
+	var restConfig *restclient.Config
+	if config.Server == "" && config.Kubeconfig == "" {
```

Comment:
Why do I need server *and* kubeconfig? I thought kubeconfig included the server?